buffer.go

```diff
@@ -103,14 +106,61 @@ func (b *Buffer) Serialize(serializationType SerializationType) ([]byte, error)
 	case TILEDB_CAPNP:
 		// The entire byte array contains Cap'nP data. Don't bother it.
 	case TILEDB_JSON:
-		// The data is a null-terminated string. Strip off the terminator.
+		// The data might be a null-terminated string. Strip off the terminator.
 		bs = bytes.TrimSuffix(bs, []byte{0})
 	default:
 		return nil, fmt.Errorf("unsupported serialization type: %v", serializationType)
 	}
 	return bs, nil
 }
 
+// WriteTo writes the contents of a Buffer to an io.Writer.
+func (b *Buffer) WriteTo(w io.Writer) (int64, error) {
+	var cbuffer unsafe.Pointer
+	var csize C.uint64_t
+
+	ret := C.tiledb_buffer_get_data(b.context.tiledbContext, b.tiledbBuffer, &cbuffer, &csize)
+	if ret != C.TILEDB_OK {
+		return 0, fmt.Errorf("error getting tiledb buffer data: %w", b.context.LastError())
+	}
+
+	if cbuffer == nil || csize == 0 {
+		return 0, nil
+	}
+
+	remaining := int64(csize)
+
+	// Because io.Writer supports writing up to 2GB of data at a time, we have to use a loop
+	// for the bigger buffers.
+	for remaining > 0 {
+		// TODO: Use min on Go 1.21+
+		var writeSize int32
+		if remaining > math.MaxInt32 {
+			writeSize = math.MaxInt32
+		} else {
+			writeSize = int32(remaining)
+		}
+
+		// Construct a slice from the buffer's data without copying it.
+		// Keep the buffer alive during the write, to prevent the GC from
+		// collecting the memory while it's being used.
+		n, err := w.Write(unsafe.Slice((*byte)(cbuffer), writeSize))
+		runtime.KeepAlive(b)
```

Comment:
I think since `b` is the object that implements this method the keepalive isn't needed

Reply:
Removed. I was being extra cautious, but if we call `b.WriteTo(w)`, it becomes clear that `b` is being kept somewhere reachable.